components/hab/src/command/pkg/binlink.rs

```diff
@@ -106,6 +118,22 @@ where
             if bin_file.file_type()?.is_dir() {
                 continue;
             }
+            if cfg!(target_os = "windows") {
+                match bin_file.path().extension() {
+                    Some(ext) => match env::var_os("PATHEXT") {
```

Comment:
I think it's hard to infer the meaning of this code without already knowing what `PATHEXT` signifies. Perhaps instead of `ext`, calling it `executable_extensions` or something like that would help.